app/models/docker_builder_service.rb

```diff
@@ -98,4 +102,13 @@ def send_after_notifications
     Samson::Hooks.fire(:after_docker_build, build)
     SseRailsEngine.send_event('builds', { type: 'finish', build: BuildSerializer.new(build, root: nil) })
   end
+
+  def registry_credentials
+    return nil unless ENV['DOCKER_REGISTRY_USER'].present? || ENV['DOCKER_REGISTRY_EMAIL'].present?
```

Comment:
It looks like Docker requires both the username and email to be set.  Even GCE which doesn't use the email field says to just fill it with a random string.